tracer/src/Datadog.Trace.Tools.Analyzers/ConfigurationAnalyzers/EnvironmentGetEnvironmentVariableAnalyzer.cs

```diff
@@ -0,0 +1,255 @@
+        /// List of methods to check. Add new entries here to extend the analyzer.
+        /// Format: ("Fully.Qualified.TypeName", "MethodName")
+        /// </summary>
+        private static readonly (string TypeName, string MethodName)[] TargetMethods =
+        {
+            ("Datadog.Trace.Util.EnvironmentHelpers", "GetEnvironmentVariable"),
+            ("Datadog.Trace.Util.EnvironmentHelpers", "EnvironmentVariableExists"),
+            ("Datadog.Trace.Util.EnvironmentHelpersNoLogging", "GetEnvironmentVariable"),
+            ("Datadog.Trace.Util.EnvironmentHelpersNoLogging", "TryCheckEnvVar"),
+            ("Datadog.Trace.Ci.CiEnvironment.IValueProvider", "GetValue"),
+        };
+
+        /// <summary>
+        /// Gets the supported diagnostics
+        /// </summary>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(UseConfigurationKeysRule, UseConfigurationKeysNotVariablesRule);
+
+        /// <summary>
+        /// Initialize the analyzer
+        /// </summary>
+        /// <param name="context">context</param>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(AnalyzeInvocationExpression, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeInvocationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+
+            // Get the method being called
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(invocation);
+            if (symbolInfo.Symbol is not IMethodSymbol calledMethod)
+            {
+                return;
+            }
+
+            // Check if this is one of our target methods
+            if (!IsTargetMethod(calledMethod))
+            {
+                return;
+            }
+
+            // Check the first argument (the environment variable name)
+            var argumentList = invocation.ArgumentList;
+            if (argumentList?.Arguments.Count > 0)
+            {
+                var argument = argumentList.Arguments[0];
+                AnalyzeEnvironmentVariableArgument(context, argument);
+            }
+        }
+
+        /// <summary>
+        /// Checks if a method is one of our target methods to analyze.
+        /// </summary>
+        private static bool IsTargetMethod(IMethodSymbol method)
+        {
+            var methodName = method.Name;
+
+            foreach (var (targetType, targetMethod) in TargetMethods)
+            {
+                if (methodName != targetMethod)
+                {
+                    continue;
+                }
+
+                // Check if the containing type matches directly
+                var containingType = method.ContainingType?.ToDisplayString();
```

Comment:
Calling `ToDisplayString()` allocates, and is relatively expensive. Ideally we should "find" the symbols and pass them through ourselves. e.g. you can see and example of that here: https://github.com/andrewlock/NetEscapades.EnumGenerators/blob/5d92a2b4a62a4b13f90b29ca8fa52ef8ac000dc6/src/NetEscapades.EnumGenerators/Diagnostics/UsageAnalyzers/GetNamesAnalyzer.cs#L32